Civi/Api4/Service/Spec/Provider/TagFieldSpecProvider.php

```diff
@@ -14,25 +14,65 @@
 
 use Civi\Api4\Query\Api4SelectQuery;
 use Civi\Api4\Service\Spec\FieldSpec;
+use Civi\Api4\Service\Spec\Provider\Generic\SpecProviderInterface;
 use Civi\Api4\Service\Spec\RequestSpec;
 use Civi\Api4\Utils\CoreUtil;
+use Civi\Core\Event\PostEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 
 /**
  * @service
  * @internal
  */
-class EntityTagFilterSpecProvider extends \Civi\Core\Service\AutoService implements Generic\SpecProviderInterface {
+class TagFieldSpecProvider extends \Civi\Core\Service\AutoService implements SpecProviderInterface, EventSubscriberInterface {
+
+  public static function getSubscribedEvents(): array {
+    return [
+      'hook_civicrm_post' => 'saveTags',
+    ];
+  }
+
+  public function saveTags(PostEvent $e) {
+    if (empty($e->params['tags'])) {
```

Comment:
For entities other than Contacts, this seems to work fine.

But for Contacts the `params` don't seem to be passed to the post hook. Is that expected?